File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    first = [1,2,3,4,5]
    second = [2,1,2,3,2,4,2,5]
    third = [3,3,1,1,2,2,4,4,5,5]
    
    result = [0 for i in range(3)]
    
    for i in range(len(answers)):
        if answers[i] == first[i%len(first)]:
            result[0] += 1
        if answers[i] == second[i%len(second)]:
            result[1] += 1
        if answers[i] == third[i%len(third)]:
            result[2] += 1
    
    for j in range(len(result)):
        if max(result) == result[j]:
            answer.append(j+1)
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] std1 = {1, 2, 3, 4, 5};
        int[] std2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] std3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int std1_count = 0;
        int std2_count = 0;
        int std3_count = 0;
        
        for(int i = 0; i < answers.length; i++) {
            if (answers[i] == std1[i % std1.length]) std1_count++;
            if (answers[i] == std2[i % std2.length]) std2_count++;
            if (answers[i] == std3[i % std3.length]) std3_count++;
        }
        
        int maxScore = Math.max(std1_count, Math.max(std2_count, std3_count));
        ArrayList<Integer> result = new ArrayList<>();
        if (std1_count == maxScore) result.add(1);
        if (std2_count == maxScore) result.add(2);
        if (std3_count == maxScore) result.add(3);
        
        answer = result.stream().mapToInt(i -> i).toArray();
        
        return answer;
    }
}
